packages/xchain-haven/src/client.ts

```diff
@@ -0,0 +1,315 @@
+        : []
+      const to: Array<TxTo> = isIn
+        ? [
+            {
+              amount: baseAmount(havenTx.toAmount, 12),
+              asset: getAssetByTicker(havenTx.to_asset_type),
+              to: havenTx.hash,
+            },
+          ]
+        : []
+
+      const tx: Tx = {
+        hash: havenTx.hash,
+        date: new Date(havenTx.timestamp),
+        type: havenTx.from_asset_type === havenTx.to_asset_type ? TxType.Transfer : TxType.Unknown,
+        from,
+        to,
+        asset: isOut ? getAssetByTicker(havenTx.from_asset_type) : getAssetByTicker(havenTx.to_asset_type),
+      }
+
+      return tx
+    })
+
+    const txPages: TxsPage = {
+      txs,
+      total: txs.length,
+    }
+    return txPages
+  }
+  async getTransactionData(txId: string, _assetAddress?: string): Promise<Tx> {
+    const havenTx = await this.havenSDK.getTx(txId)
+
+    const isOut: boolean = baseAmount(havenTx.total_sent[havenTx.from_asset_type], 12).gt('0')
+    const isIn: boolean = baseAmount(havenTx.total_received[havenTx.to_asset_type], 12).gt('0')
+
+    const from: Array<TxFrom> = isOut
+      ? [
+          {
+            amount: baseAmount(havenTx.fromAmount, 12),
+            asset: getAssetByTicker(havenTx.from_asset_type),
+            from: havenTx.hash,
+          },
+        ]
+      : []
+    const to: Array<TxTo> = isIn
+      ? [
+          {
+            amount: baseAmount(havenTx.toAmount, 12),
+            asset: getAssetByTicker(havenTx.to_asset_type),
+            to: havenTx.hash,
+          },
+        ]
+      : []
+
+    const asset: Asset = isOut ? getAssetByTicker(havenTx.from_asset_type) : getAssetByTicker(havenTx.to_asset_type)
+
+    const tx: Tx = {
+      hash: havenTx.hash,
+      date: new Date(havenTx.timestamp),
+      type: isIn && isOut ? TxType.Unknown : TxType.Transfer,
+      from,
+      to,
+      asset,
+    }
+    return tx
+  }
+
+  async transfer(params: TxParams): Promise<TxHash> {
+    const { amount, asset, recipient, memo } = params
+    if (asset === undefined) throw new Error('please specify asset it in Client.transfer() for Haven')
+    const amountString = amount.amount().toString()
+    let txHash
+    try {
+      txHash = await this.havenSDK.transfer(amountString, asset.ticker as HavenTicker, recipient, memo)
+      return txHash
+    } catch (e) {
+      throw new Error('Tx could not be sent')
+    }
+  }
+
+  purgeClient(): void {
+    super.purgeClient()
+    this.havenSDK.purge()
+    this.havenMnemonic = undefined
+  }
+
+  /**
+   * checks if syncing is going on which can take up to a few seconds or hours
+   * @returns boolean
+   */
+  async isSyncing(): Promise<boolean> {
+    return this.havenSDK.isSyncing()
+  }
+
+  /**
+   * subscribe sync progress which updates every X seconds with SyncStats
+   * and notifys on completion
+   * @param observer
+   */
+  subscribeSyncProgress(observer: SyncObserver): void {
+    this.havenSDK.subscribeSyncProgress(observer)
```

Comment:
Is there a way to unsubscribe an observer if needed (e.g. in `purgeClient`)?

Reply:
added a unsubscribe method, when `purgeClient()` is called all observers are removed